src/common/misc.c

```diff
@@ -40,7 +40,7 @@ daos_sgl_init(d_sg_list_t *sgl, unsigned int nr)
 
 	sgl->sg_nr = nr;
 	if (nr == 0)
-		return 0;
+		return -DER_INVAL;
```

Comment:
Why is this invalid?

Reply:
The reason is callers of daos_sgl_init() asserts rc==0, then dereference sgl.sg_iovs which was set to NULL in line 39 here.

Reply:
Code should not be using assert to check that daos_sgl_init() passes, if it is (and it's not test code) then it needs to be fixed.

Reply:
Right, seems like all code asserting this rc are test code - the Coverity issue is about the test code, for nr==0 case some test code will dereference NULL point, should we keep the daos_sgl_init code and change all test code to not use assert?
Adding @johannlombardi seems like added this line:
d75e528d src/common/misc.c             (Johann Lombardi       2016-11-09 12:24:49 +0000  39)    memset(sgl, 0, sizeof(*sgl));